lib/express_validator.js

```diff
@@ -259,13 +263,32 @@ var expressValidator = function(options) {
  */
 
 function validateSchema(schema, req, loc, options) {
+  var locations = ['body', 'params', 'query'],
+    currentLoc = loc;
+
   for (var param in schema) {
-    loc = loc === 'any' ? locate(req, param) : loc;
-    var validator = new ValidatorChain(param, null, req, loc, options);
+
+    // check if schema has defined location
+    if (schema[param].hasOwnProperty('in')) {
+      if (locations.indexOf(schema[param].in) !== -1) {
+        currentLoc = schema[param].in;
+      } else {
+        // skip params where defined location is not supported
+        continue;
+      }
+    }
+
+    currentLoc = currentLoc === 'any' ? locate(req, param) : currentLoc;
+    var validator = new ValidatorChain(param, null, req, currentLoc, options);
     var paramErrorMessage = schema[param].errorMessage;
     delete schema[param].errorMessage;
 
     for (var methodName in schema[param]) {
+      // skip method if this is location definition
+      if (methodName === 'in') {
+        currentLoc = loc;
```

Comment:
Maybe I'm missing something, but why are we redefining `currentLoc` here? Shouldn't it be finalized on line 274?